File: src/test/java/cz/vhromada/catalog/web/mapper/SeasonMapperTest.java
package cz.vhromada.catalog.web.mapper;

import static org.assertj.core.api.Assertions.assertThat;

import cz.vhromada.catalog.entity.Season;
import cz.vhromada.catalog.web.common.SeasonUtils;
import cz.vhromada.catalog.web.fo.SeasonFO;

import org.junit.jupiter.api.BeforeEach;
import org.junit.jupiter.api.Test;
import org.mapstruct.factory.Mappers;

/**
 * A class represents test for mapper between {@link Season} and {@link SeasonFO}.
 *
 * @author Vladimir Hromada
 */
class SeasonMapperTest {

    /**
     * Mapper for seasons
     */
    private SeasonMapper mapper;

    /**
     * Initializes mapper.
     */
    @BeforeEach
    void setUp() {
        mapper = Mappers.getMapper(SeasonMapper.class);
    }

    /**
     * Test method for {@link SeasonMapper#map(Season)}.
     */
    @Test
    void map() {
        final Season season = SeasonUtils.getSeason();

        final SeasonFO seasonFO = mapper.map(season);

        SeasonUtils.assertSeasonDeepEquals(seasonFO, season);
    }

    /**
     * Test method for {@link SeasonMapper#map(Season)} with null season.
     */
    @Test
    void map_NullSeason() {
        assertThat(mapper.map(null)).isNull();
    }

    /**
     * Test method for {@link SeasonMapper#mapBack(SeasonFO)}.
     */
    @Test
    void mapBack() {
        final SeasonFO seasonFO = SeasonUtils.getSeasonFO();

        final Season season = mapper.mapBack(seasonFO);

        SeasonUtils.assertSeasonDeepEquals(seasonFO, season);
    }

    /**
     * Test method for {@link SeasonMapper#mapBack(SeasonFO)} with null season.
     */
    @Test
    void mapBack_NullSeason() {
        assertThat(mapper.mapBack(null)).isNull();
    }

}

File: src/test/kotlin/cz/vhromada/catalog/web/mapper/SeasonMapperTest.kt
package cz.vhromada.catalog.web.mapper

import cz.vhromada.catalog.entity.Season
import cz.vhromada.catalog.web.CatalogMapperTestConfiguration
import cz.vhromada.catalog.web.common.SeasonUtils
import cz.vhromada.catalog.web.fo.SeasonFO
import org.junit.jupiter.api.Test
import org.junit.jupiter.api.extension.ExtendWith
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.test.context.ContextConfiguration
import org.springframework.test.context.junit.jupiter.SpringExtension

/**
 * A class represents test for mapper between [Season] and [SeasonFO].
 *
 * @author Vladimir Hromada
 */
@ExtendWith(SpringExtension::class)
@ContextConfiguration(classes = [CatalogMapperTestConfiguration::class])
class SeasonMapperTest {

    /**
     * Mapper for seasons
     */
    @Autowired
    private lateinit var mapper: SeasonMapper

    /**
     * Test method for [SeasonMapper.map].
     */
    @Test
    fun map() {
        val season = SeasonUtils.getSeason()

        val seasonFO = mapper.map(season)

        SeasonUtils.assertSeasonDeepEquals(seasonFO, season)
    }

    /**
     * Test method for [SeasonMapper.mapBack].
     */
    @Test
    fun mapBack() {
        val seasonFO = SeasonUtils.getSeasonFO()

        val season = mapper.mapBack(seasonFO)

        SeasonUtils.assertSeasonDeepEquals(seasonFO, season)
    }

}
